tower-spawn-ready/src/service.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{
+    error::{Error, SpawnError},
+    future::{BackgroundReady, BackgroundReadyExecutor},
+};
+use futures::{future, try_ready, Async, Future, Poll};
+use std::marker::PhantomData;
+use tokio_executor::DefaultExecutor;
+use tokio_sync::oneshot;
+use tower_service::Service;
+
+/// Spawns tasks to drive an inner service to readiness.
+///
+/// See crate level documentation for more details.
+pub struct SpawnReady<T, Request, E>
+where
+    T: Service<Request>,
+    T::Error: Into<Error>,
+    E: BackgroundReadyExecutor<T, Request>,
+{
+    executor: E,
+    inner: Inner<T>,
+    _marker: PhantomData<fn(Request)>,
+}
+
+enum Inner<T> {
+    Service(Option<T>),
+    Future(oneshot::Receiver<Result<T, Error>>),
+}
+
+impl<T, Request> SpawnReady<T, Request, DefaultExecutor>
+where
+    T: Service<Request> + Send + 'static,
+    T::Error: Into<Error>,
+    Request: 'static,
+{
+    /// Creates a new `SpawnReady` wrapping `service`.
+    ///
+    /// The default Tokio executor is used to drive service readiness, which
+    /// means that this method must be called while on the Tokio runtime.
+    pub fn new(service: T) -> Self {
+        Self::with_executor(service, DefaultExecutor::current())
+    }
+}
+
+impl<T, Request, E> SpawnReady<T, Request, E>
+where
+    T: Service<Request> + Send,
+    T::Error: Into<Error>,
+    E: BackgroundReadyExecutor<T, Request>,
+{
+    /// Creates a new `SpawnReady` wrapping `service`.
+    ///
+    /// `executor` is used to spawn a new `BackgroundReady` task that is
+    /// dedicated to driving the inner service to readiness.
+    ///
+    /// `bound` gives the maximal number of requests that can be queued for the service before
```

Comment:
It looks like this no longer takes a parameter called `bound` --- is this comment stale?